### PR TITLE
feat: create a pr template for code

### PR DESCRIPTION
**Pull request by K-Beicher**
This is the first draft of the PR Template for code

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/pull-request-code.qmd
Comment (signekb) [inline .github/PULL_REQUEST_TEMPLATE/pull-request-code.qmd:1]: I also think this needs to be a .md (and not .qmd)? :)

File: .github/PULL_REQUEST_TEMPLATE/pull-request-code.qmd [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+## Testing
+
+- [ ] Yes
+- [ ] No, not needed (give a reason below)
+- [ ] No, I need help writing them
+
+<!-- Please explain why the tests are not needed for this PR here -->
+
+## Hardware and software configuration
```
Comment (signekb) [inline .github/PULL_REQUEST_TEMPLATE/pull-request-code.qmd:37]: Go for title casing in all headers? I.e., capitalise all  words except for minor words that are not the first or last word of the title?

File: .github/PULL_REQUEST_TEMPLATE/pull-request-code.qmd [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+## Code documentation
+
+<!-- Please detail which parts, if any, of the documentation has been updated -->
+
+## Reviewer focus
+
+<!-- Any particular section the reviewer should focus on, anywhere that would be a good place to start? -->
+
+### Code Author Checklist
+<!-- This is to help you determine if your work is ready to be reviewed -->
```
Comment (signekb) [inline .github/PULL_REQUEST_TEMPLATE/pull-request-code.qmd:50]: Again: what to do if any of there are irrelevant? E.g., should you check the first box below if you haven't included any new automated tests?

File: .github/PULL_REQUEST_TEMPLATE/pull-request-code.qmd [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+- [ ] Automated tests are added where needed
+- [ ] All (new and existing) tests have run without falling over
```
Comment (signekb) [inline .github/PULL_REQUEST_TEMPLATE/pull-request-code.qmd:52]: run locally?

File: .github/PULL_REQUEST_TEMPLATE/pull-request-code.qmd
Comment (K-Beicher) [reply .github/PULL_REQUEST_TEMPLATE/pull-request-code.qmd:1]: I'm going to change them once we are happy with the changes, I like the code snippets too much :-)

**Review by signekb — APPROVED** [review]
NICE! 🎉

**Review by lwjohnst86 — APPROVED** [review]
I removed the section on hardware/software since I don't feel it is relevant for us (right now anyway). I revised and simplified some of the wording and reformatted the text a bit. But mostly, thanks! Will merge in now :grin:

**Pull request merged**